src/batch_edge_query.js

```diff
@@ -7,10 +7,11 @@ const utils = require('./utils');
 const LogEntry = require('./log_entry');
 
 module.exports = class BatchEdgeQueryHandler {
-  constructor(kg, resolveOutputIDs = true) {
+  constructor(kg, caching, resolveOutputIDs = true) {
```

Comment:
wonder if it's better to change the signature to `constructor(kg, resolveOutputIDs = true, options=null)`:
* it's compatible with the previous signature `constructor(kg, resolveOutputIDs = true)`
* `options` can be an optional object reserved to pass any additional parameters, including caching)

Reply:
Yes, I think that would be better, I will change it.